File: YuGiOh2/Cards/C83764718.cs
using System.Linq;
using YuGiOh2.Base;
using YuGiOh2.Data;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 死者苏生
    /// </summary>
    public class C83764718
    {
        public static int Type { get; } = (int)ChooseTargetType.AllMonsterInGrave;

        public static bool CheckIfAvailable(Card card, Player player, Player enemy)
        {
            return (enemy.Grave.Any(c => c.CardCategory == 0) || player.Grave.Any(c => c.CardCategory == 0)) &&
                player.Field.MonsterFields.Any(c => c == null);
        }

        public static void ProcessEffect(Card card, string targetID, Player player, Player enemy)
        {
            if (targetID == null)
                return;

            int[] sort = new int[] { 2, 1, 3, 0, 4 };
            foreach (var num in sort)
            {
                if (player.Field.MonsterFields[num] != null)
                    continue;

                MonsterCard monster = (player.Grave.FirstOrDefault(c => c.UID == targetID) ?? 
                    enemy.Grave.FirstOrDefault(c => c.UID == targetID)) as MonsterCard;
                if (monster == null)
                    return;

                player.Grave.Remove(monster);
                enemy.Grave.Remove(monster);
                player.Hands.Add(monster);
                player.CanSummon = true;
                player.SummonMonsterFromHands(monster.UID);
                monster.Status.CanChangePosition = true;
            }
        }
    }
}

File: YuGiOh2/Scripts/C83764718.lua
Type = 25;
--死者苏生
function checkIfAvailable(player)
    if #player.Field.MonsterFields == 5 then
        return false;
    end
    for i, card in pairs(player.Enemy.Grave) do
        if card.CardCategory == 0 then
            return true;
        end
    end
    for i, card in pairs(player.Grave) do
        if card.CardCategory == 0 then
            return true;
        end
    end
    return false;
end

function processEffect(targetID, player)
    if targetID == nil then
        return ;
    end
    local target;
    local enemyGrave = {};
    for i, card in pairs(player.Enemy.Grave) do
        if card.CardCategory == 0 and card.UID == targetID then
            target = card;
        else
            table.insert(enemyGrave, card);
        end
    end
    player.Enemy.Grave = enemyGrave;
    if target == nil then
        local playerGrave = {};
        for i, card in pairs(player.Grave) do
            if card.CardCategory == 0 and card.UID == targetID then
                target = card;
            else
                table.insert(playerGrave, card);
            end
        end
        player.Grave = playerGrave;
    end
    if target == nil then
        return ;
    end
    if #player.Field.MonsterFields == 5 then
        table.insert(player.Grave, target);
        return ;
    end
    local tmp = player.CanSummon;
    player.CanSummon = true;
    local hands = {};
    for i, card in pairs(player.Hands) do
        table.insert(hands, card);
    end
    table.insert(hands, target);
    player.Hands = hands;
    player:SummonMonsterFromHands(target.UID);
    player:ProcessContinuousEffectWhenSummon(target.UID);
    player.Enemy:ProcessContinuousEffectWhenSummon(target.UID);
    target.Status.CanChangePosition = true;
    player.CanSummon = tmp;
end